cogito/put.go

```diff
@@ -277,14 +278,32 @@ type gitURL struct {
 	Repo  string
 }
 
+// safeParseUrl parse the given URL and return an error if the URL is not well formated.
+// this avoid to eventually leak passwords in case of error
+//
+// From https://github.com/golang/go/issues/53993
+func safeParse(rawURL string) (*url.URL, error) {
+	parsedUrl, err := url.Parse(rawURL)
+	if err != nil {
+		// check if the returned error is actually of type url.Error
+		if uerr := new(url.Error); errors.As(err, &uerr) {
+			// return only the Err and not the URL to not leak potential passwords
+			return nil, uerr.Err
+		}
+		// default error
+		return nil, errors.New("invalid URL")
```

Comment:
When we reach this point, are we 100% sure that the URL was invalid?

Reply:
This means that the URL parsing went wrong and the function `url.Parse` returned a non-nil error.
Maybe I could change the error message with `unable to parse URL` instead of `Invalid URL`?

Reply:
> When we reach this point, are we 100% sure that the URL was invalid?

Actually this point (line 292) is currently unreachable, since url.Parse (for the time being) always returns type url.Error.

But to answer the question: yes, we are 100% sure, because it is still under the error condition of url.Parse (see line 287 where the branch begins).